### PR TITLE
#201 Added "Add task" button

### LINKED ISSUE #201: Task creator has no "create" button
screenshot_2018-10-23 create task - taskcluster

It really does, but it's hidden in the hamburger menu way down in the lower-right, along with "reset editor" and "update timestamps".  I think "create" is really the "OK" button for this page, and shouldn't be hidden behind a click/hover[1].  "Reset" is the "cancel" button, and "update timestamps" is sort of a utility function.  I've no idea how best to show the last one.

[1] Actually, it appears to be hover only.  When I go to click on that hamburger it enticingly shows me the options above it, then they disappear.  I just sort of mindlessly did this three times until I stopped and thought about it..

### PR DESCRIPTION
**Pull request by sammysamx20**
Fixes #201.

Not completed yet, I wanted you to check code style if you can!

### CODE REVIEW

**Comment by sammysamx20** [comment]
@helfi92 Strange it should've worked... I'll push again and see whats going on.

**Comment by sammysamx20** [comment]
@helfi92 It passes!

**Comment by helfi92** [comment]
How is it going @sammysamx20? Let me know if I can help :)

**Comment by sammysamx20** [comment]
Hey sorry I got busy this last weekend! I should be able to work on this today

**Comment by helfi92** [comment]
No worries! Just wanted to see if everything was okay :)

**Comment by sammysamx20** [comment]
Actually I do have one question, I was planning to wrap the buttons in a div. But I was wondering how you got the hamburger menu down at the bottom right and where the CSS is for that?

**Comment by helfi92** [comment]
> I was planning to wrap the buttons in a div

Yes, you can definitely wrap the button in a div. Something like that:

```jsx
<Tooltip title="Create Task"> 
  <div>
    <Button 
      variant="fab" 
       // ...
    > 
      <PlusIcon /> 
    </Button> 
  <div>
</Tooltip> 
```

> But I was wondering how you got the hamburger menu down at the bottom right and where the CSS is for that?

To have the button have the "circle" shape, you have to provide the `variant="fab"` prop. In order to have it positioned in the bottom right, you'll need to extend it with a class. Because we do this in many places in our app, we have the following class that you can use: https://github.com/taskcluster/taskcluster-web/blob/master/src/theme.js#L114-L121. You can refer to it via `theme.mixins.fab`. For example, it is done in https://github.com/taskcluster/taskcluster-web/blob/0a187e352bf6eb97c60682ba69b4004dd5cba632/src/components/SecretForm/index.jsx#L26-L28.

Hope that helps.

**Comment by sammysamx20** [comment]
@helfi92 Sorry for the long delay here's what I have!

**Comment by helfi92** [comment]
Do you mind rebasing on master, otherwise it gets hard to review this patch. There's a couple of ways to do this but I usually do:

```bash
# step 1: from your branch, run
git pull https://github.com/taskcluster/taskcluster-web.git master --rebase

# step 2: if there's a file conflict, fix it, then git add. Notice that you don't do git commit after git add here:
git add <file-with-conflict>
git rebase --continue

# step 3: if there's another conflict, go back to step 2

# step 4:  force push your change
git push origin <your-branch-name> --force
```

**Comment by sammysamx20** [comment]
Should work now with rebase @helfi92

File: src/views/Tasks/CreateTask/index.jsx [hunk 1]
```diff
@@ -277,18 +278,24 @@ export default class CreateTask extends Component {
...
-                  requiresAuth
-                  tooltipOpen
-                  icon={<PlusIcon />}
+              <Tooltip
+                style={{ position: 'fixed', bottom: 16, right: 90 }}
+                placement="left"
+                title="Create Task"
+              >
+                <Button
+                  color="secondary"
```
Comment (helfi92) [inline src/views/Tasks/CreateTask/index.jsx]: Remove `color`. The color is already specified in `classes.createIcon`.

**Comment by sammysamx20** [comment]
Let me know if this works.

File: src/views/Tasks/CreateTask/index.jsx [hunk 2]
```diff
@@ -61,6 +63,12 @@ const defaultTask = {
   createIcon: {
     ...theme.mixins.successIcon,
   },
+  toolTip: {
+    position: 'fixed',
+    bottom: 16,
```
Comment (helfi92) [inline src/views/Tasks/CreateTask/index.jsx]: `bottom: theme.spacing.double`,

File: src/views/Tasks/CreateTask/index.jsx [hunk 3]
```diff
@@ -61,6 +63,12 @@ const defaultTask = {
...
+    right: 90,
```
Comment (helfi92) [inline src/views/Tasks/CreateTask/index.jsx]: `right: theme.spacing.unit * 11`,

File: src/views/Tasks/CreateTask/index.jsx [hunk 4]
```diff
@@ -61,6 +63,12 @@ const defaultTask = {
...
+    background: '#4CAF50',
```
Comment (helfi92) [inline src/views/Tasks/CreateTask/index.jsx]: Remove this.

**Comment by sammysamx20** [comment]
Actually I think I figured it out

File: src/views/Tasks/CreateTask/index.jsx [hunk 5]
```diff
@@ -277,18 +282,22 @@ export default class CreateTask extends Component {
...
                 onChange={this.handleTaskChange}
               />
-              <SpeedDial>
-                <SpeedDialAction
-                  requiresAuth
-                  tooltipOpen
-                  icon={<PlusIcon />}
+              <Tooltip
+                className={classes.createIcon}
+                placement="left"
```
Comment (helfi92) [inline src/views/Tasks/CreateTask/index.jsx]: `placement="top"`.

File: src/views/Tasks/CreateTask/index.jsx [hunk 6]
```diff
@@ -277,18 +282,22 @@ export default class CreateTask extends Component {
...
+                title="Create Task"
+              >
+                <Button
+                  variant="fab"
```
Comment (helfi92) [inline src/views/Tasks/CreateTask/index.jsx]: Could you also add `requiresAuth`, please. We don't user to create tasks when signed out.

```jsx
<Button
  requiresAuth
  variant="fab"
  // ...
```

**Comment by sammysamx20** [comment]
Do you know what is causing the light mode to turn it to black?

**Comment by sammysamx20** [comment]
Actually I think I got it

**Comment by helfi92** [comment]
It's probably https://github.com/taskcluster/taskcluster-web/blob/8d1015b74a84dde4703988433b98043326cc25f0/src/App/Main.jsx#L21-L23. In `Main.jsx`, you can create another style `fabIcon` with the following style:

```diff
+    fabIcon: {
+      '& .mdi-icon': {
+        fill: 'white',
+      },
+    },
```

Then from the creat task view, you can do:

```jsx
  createIcon: {
    ...theme.mixins.successIcon,
    ...theme.mixins.fabIcon,
    position: 'fixed',
    bottom: theme.spacing.double,
    right: theme.spacing.unit * 11,
},
```

I think that should fix it.

**Comment by sammysamx20** [comment]
I pushed some changes let me know if it works!

**Comment by sammysamx20** [comment]
is this okay?

**Comment by sammysamx20** [comment]
phew I'm glad it works thanks for the help!!

**Comment by helfi92** [comment]
No problem. Awesome work!

File: src/views/Tasks/CreateTask/index.jsx [hunk 7]
```diff
@@ -277,18 +283,20 @@ export default class CreateTask extends Component {
...
                 onChange={this.handleTaskChange}
               />
-              <SpeedDial>
-                <SpeedDialAction
-                  requiresAuth
-                  tooltipOpen
-                  icon={<PlusIcon />}
+              <Tooltip title="Create Task">
+                <Button
+                  requireAuth
```
Comment (helfi92) [inline src/views/Tasks/CreateTask/index.jsx]: Oops. This needs to be `requiresAuth` 😅

Comment (sammysamx20) [reply src/views/Tasks/CreateTask/index.jsx]: Aw man I'll do that now :(

**Pull request merged**